File: SampleCheck.java
public class BadCodeExample {

    static int x = 0;
    static String data = "Hello ";
    static String unusedGlobal = "I am useless";

    public static void main(String[] args) {

        int a = 10;
        int b = 0;
        int result = 0;

        // Unused variable
        String temp = "not used";

        // Null pointer risk
        String str = null;
        if (str.equals("test ")) {
            System.out.println("Match ");
        }

        // Division by zero risk
        try {
            result = a / b; // ❌ runtime error
        } catch (Exception e) {

        }


        String username = "admin";
        String password = "123456 ";

        int i = 0;
        while (i < 5) {
            System.out.println(i);

        }

        // Always true condition
        if (true) {
            System.out.println("This block always runs");
        }

        // Magic numbers
        if (a > 7) {
            System.out.println("Magic number used");
        }

        // Bad naming
        int q = 5;
        int w = 10;
        int e = q + w;
        System.out.println(e);

        // Duplicate code
        System.out.println("Duplicate");
        System.out.println("Duplicate");

        // Long method doing too many things
        doEverything();
    }

    public static void doEverything() {
        System.out.println("Doing everything...");
        int x = 10;
        int y = 20;
        int z = x + y;
        System.out.println(z);

        // More duplicate logic
        int x2 = 10;
        int y2 = 20;
        int z2 = x2 + y2;
        System.out.println(z2);
    }
}

File: sample_check.py
import os
import math

GLOBAL_VAR = "I should not be global"
another_global = 999


def do_stuff():

    x = 100


    a = 5
    b = 10
    c = a + b
    print(c)


    list = [1, 2, 3]  
    for i in range(0, len(list)):  # ❌ not pythonic
        print(list[i])

    # Hardcoded sensitive data
    password = "admin123"
    api_key = "XYZ-SECRET-KEY"

    # Always true condition
    if True:
        print("This always runs")

    # Try/except bad practice
    try:
        value = int("abc")  # ❌ will fail
    except:
        pass  # ❌ swallowing error silently

    # Division by zero risk
    num = 10
    denom = 0
    try:
        print(num / denom)
    except Exception as e:
        print("error")  # ❌ not logging actual error

    # Infinite loop
    i = 0
    while i < 3:
        print("Looping...")
        # ❌ forgot i += 1

    # Duplicate code
    print("repeat")
    print("repeat")

    # Poor function design (too many responsibilities)
    messy_function()


def messy_function():
    # Duplicate logic
    x = 1 + 2
    print(x)

    x = 1 + 2
    print(x)

    # Magic numbers
    if x > 3:
        print("Magic number used")

    # None comparison (bad style)
    y = None
    if y == None:  # ❌ should use 'is None'
        print("y is None")


if __name__ == "__main__":
    do_stuff()